### PR TITLE
refactor(cli): enhance plugin subcommand, closes #7749

### LINKED ISSUE #7749: [feat] Add CLI option to `tauri plugin init` to enable/disable iOS/Android projects
### Describe the problem

Right now when running `tauri plugin init` it will always create a respective Swift and Kotlin project. To make a desktop-only plugin this is an awkward experience as the developer would have to then remove the respective projects.

### Describe the solution you'd like

I personally prefer opt-in, so I'd see the default of `tauri plugin init` to NOT create a mobile project (or to prompt the dev if they'd like to). This could give the option to add flags such as `tauri plugin init [--with-ios|--with-andoird]` to bypass that prompt.

I'm not sure if running `tauri plugin init --with-ios` would still prompt the user if they'd like to init android, or if that would only init the base desktop project with iOS and skip android completely. 

### Alternatives considered

_No response_

### Additional context

Originally brought up in https://github.com/tauri-apps/tauri-docs/pull/1412/files/be2b0c2e42ca05842674bdb5cd5cc55e1f551596#diff-2e62f026cfd52587aa884598003549fdb22855e8637da24c74c0ece364922bd8R36

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [X] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [X] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [X] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [X] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [X] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: .changes/cli-plugin-init.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+'tauri-cli': 'patch'
+'@tauri-apps/cli': 'patch'
```
Comment (amrbashir) [inline .changes/cli-plugin-init.md]:
```suggestion
'tauri-cli': 'patch:breaking'
'@tauri-apps/cli': 'patch:breaking'
```

File: tooling/cli/src/plugin.rs [hunk 2]
```diff
@@ -25,17 +28,46 @@ pub struct Cli {
...
   }
 
   Ok(())
 }
+
+fn infer_plugin_name<P: AsRef<Path>>(directory: P) -> Result<String> {
+  let dir = directory.as_ref();
+  let carg_toml_path = dir.join("Cargo.toml");
+  let name = if carg_toml_path.exists() {
+    let contents = std::fs::read(carg_toml_path)?;
```
Comment (lucasfernog) [inline tooling/cli/src/plugin.rs]:
```suggestion
  let cargo_toml_path = dir.join("Cargo.toml");
  let name = if cargo_toml_path.exists() {
    let contents = std::fs::read(cargo_toml_path)?;
```

File: tooling/cli/src/plugin.rs [hunk 3]
```diff
@@ -25,17 +28,46 @@ pub struct Cli {
 
 #[derive(Subcommand)]
 enum Commands {
+  New(new::Options),
```
Comment (lucasfernog) [inline tooling/cli/src/plugin.rs:31]: I prefer `create` instead of `new` what do you think?

**Review by lucasfernog — CHANGES_REQUESTED** [review]

File: tooling/cli/src/plugin.rs (hunk 3 above)
Comment (amrbashir) [reply tooling/cli/src/plugin.rs:31]: `new` is the usually the term I see in bootstrapping CLIs when you have both `init` and `new`, for example, `cargo new` and `cargo init` (I honestly can't think of another example, but I am quite certain I saw it somewhere else)

Comment (amrbashir) [reply tooling/cli/src/plugin.rs:31]: or we could have both

**Review by lucasfernog — APPROVED** [review]

File: tooling/cli/src/plugin.rs (hunk 3 above)
Comment (lucasfernog) [reply tooling/cli/src/plugin.rs:31]: sold me on the cargo example

**Pull request merged**